File: GXpert/GXpert.Web/Modules/ServerTypes/Activation/ActivationForm.ts
import { LookupEditor, ServiceLookupEditor, StringEditor, DateEditor, EnumEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EKeyStatus } from "../Web/Enums.EKeyStatus";

export interface ActivationForm {
    PlayListId: LookupEditor;
    TeacherId: LookupEditor;
    ActivationLogId: ServiceLookupEditor;
    DeviceId: StringEditor;
    DeviceDetails: StringEditor;
    ActivationDate: DateEditor;
    ExpiryDate: DateEditor;
    EStatus: EnumEditor;
}

export class ActivationForm extends PrefixedContext {
    static readonly formKey = 'Activation.Activation';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!ActivationForm.init)  {
            ActivationForm.init = true;

            var w0 = LookupEditor;
            var w1 = ServiceLookupEditor;
            var w2 = StringEditor;
            var w3 = DateEditor;
            var w4 = EnumEditor;

            initFormType(ActivationForm, [
                'PlayListId', w0,
                'TeacherId', w0,
                'ActivationLogId', w1,
                'DeviceId', w2,
                'DeviceDetails', w2,
                'ActivationDate', w3,
                'ExpiryDate', w3,
                'EStatus', w4
            ]);
        }
    }
}

[EKeyStatus]; // referenced types

File: GXpert/GXpert.Web/Modules/Activation/Activation/ActivationForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.Activation.Forms;

[FormScript("Activation.Activation")]
[BasedOnRow(typeof(ActivationRow), CheckNames = true)]
public class ActivationForm
{
    [HalfWidth]
    public int PlayListId { get; set; }
    [HalfWidth]
    public int TeacherId { get; set; }
    [HalfWidth]
    public int ActivationLogId { get; set; }
    [HalfWidth]
    public string DeviceId { get; set; }
    [HalfWidth]
    public string DeviceDetails { get; set; }
    [HalfWidth]
    public DateTime ActivationDate { get; set; }
    [HalfWidth]
    public DateTime ExpiryDate { get; set; }
    [HalfWidth]
    public short EStatus { get; set; }
}